sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueClientBuilder.java

```diff
@@ -156,7 +129,28 @@ public QueueClient buildClient() {
      * or {@link #sasToken(String) SAS token} has been set.
      */
     public QueueAsyncClient buildAsyncClient() {
-        return new QueueAsyncClient(constructImpl(), queueName, accountName);
+        Objects.requireNonNull(queueName, "'queueName' cannot be null.");
```

Comment:
iirc, you have that Utility.assertNotNull

Reply:
Done